controllers/managedocs_controller.go

```diff
@@ -1752,6 +1762,51 @@ func (r *ManagedOCSReconciler) updateMCGCSV(csv *opv1a1.ClusterServiceVersion) e
 	return nil
 }
 
+func (r *ManagedOCSReconciler) reconcileOCSConsolePlugin() error {
+	r.Log.Info("Reconciling OCSConsolePlugin")
+	if err := r.get(r.ocpConsolePlugin); err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+	if r.ocpConsolePlugin.UID == "" {
+		// ODF Operator has to bring this CR,
+		// so if OCP ConsolePlugin is not ready, we ruturn silently
+		r.Log.Info("OCSConsolePlugin CR not found, skipping console proxy configuration")
+		return nil
+	}
+	expectedProxy := templates.ConsolePluginProxyTemplate.DeepCopy()
+	expectedProxy.Service.Name = prometheusServiceName
+	expectedProxy.Service.Namespace = r.namespace
+
+	var matchingProxy *consolev1alpha1.ConsolePluginProxy
+	for i := range r.ocpConsolePlugin.Spec.Proxy {
+		currProxy := &r.ocpConsolePlugin.Spec.Proxy[i]
+		if currProxy.Service.Name == expectedProxy.Service.Name &&
+			currProxy.Service.Namespace == expectedProxy.Service.Namespace {
+			matchingProxy = currProxy
+			break
+		}
+	}
+	updateConsolePluginProxy := true
```

Comment:
You are not updating the console plugin proxy, you are updating the console plugin resource. 
Can we name this correctly?
```suggestion
	shouldUpdateConsolePlugin := true
```

Also, let's not assume that an update is needed and start with a value of "false", this will reduce the `if else` cases from 3 to 2.

Reply:
Done